app/views/messages/_message.html.erb

```diff
@@ -245,6 +245,24 @@ end
                 <% end %>
               </menu>
             </div>
+            <% if last_message %>
+              <% input_tokens = conversation.messages.sum(:input_token_count) %>
+              <% output_tokens = conversation.messages.sum(:output_token_count) %>
+
+              <div class="dropdown dropdown-top flex items-center ml-2">
+                <%= button_tag "$",
+                               tabindex: 0,
+                               role: :button,
+                               class: "text-gray-600 dark:text-gray-300 hover:text-gray-900 dark:hover:text-white",
+                               data: { role: "show-token-info" }
+                %>
+
+                <div tabindex="0" class="dropdown-content -ml-6 z-10 p-2 shadow-xl bg-base-100 rounded-box w-52 dark:!bg-gray-700">
+                  <p class="py-1 px-2 text-sm text-gray-700 dark:text-gray-300">Input tokens: <%= input_tokens %></p>
+                  <p class="py-1 px-2 text-sm text-gray-700 dark:text-gray-300">Output tokens: <%= output_tokens %></p>
+                </div>
+              </div>
+            <% end %>
```

Comment:
@lumpidu I just realized a pretty decent issue with this which I didn't consider before: the _message partial updates *a lot* because we stream continual fragments. Having this partial trigger some SQL queries doesn't seem like a good idea.

I spent a little while looking at the UI and I made room for this next to the conversation title, in the left column. Here is the screenshot:
<img width="279" alt="image" src="https://github.com/user-attachments/assets/4889543b-5b25-46ca-a0d9-23fcfe9e9c0c">

What do you think? It's this PR #463 

One implication of putting it over here is that we need to make sure it updates itself. I already have the front-end wired up with turbo so that if a conversation model changes at all, it will broadcast this update to the left side and replace the full `_conversation` partial. Every time a message finishes streaming, it does `message.conversation.touch` to trigger an update to the timestamp of the conversation object, and this also causes the conversation title to update because it has an after_touch hook (all this is within `conversation.rb`)

If you like the direction I'm heading with this, then I can merge in that PR 463 and then you can pull main back into this PR. You can then update this PR to be:

* Add 3 token columns to the conversation table: `input_token_count`, `output_token_count`, and `token_cost`
* Add an after_save hook within message which updates these three columns on the conversation
* Within get_next_ai_message_job find the conversation.touch and remove this, it's no longer necessary now that message.save always triggers the conversation to update
* Within conversation.rb change the `after_touch` to an `after_save`

Reply:
Great! It's now in main so you can pull it into this branch.